### PR TITLE
Prevent Deck's '/rerun' path from exposing info about hidden repo jobs.

### LINKED ISSUE #5910: hide pods from hidden repos
from a discussion with @cjwagner, hidden_repos is nice but probably isn't sufficient if you can guess pod IDs, deck should restrict all endpoints from showing these repos

/area prow
/kind bug

### PR DESCRIPTION
**Pull request by cjwagner**
fixes #5910 
/cc @BenTheElder

### CODE REVIEW

**Comment by BenTheElder** [comment]
Changes LGTM, thanks!
What about the log endpoint?

File: prow/cmd/deck/main.go [hunk 1]
```diff
@@ -316,16 +317,21 @@ type pjClient interface {
...
 
-func handleRerun(kc pjClient) http.HandlerFunc {
+func handleRerun(config configAgent, kc pjClient) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		name := r.URL.Query().Get("prowjob")
 		if !objReg.MatchString(name) {
 			http.Error(w, "Invalid ProwJob query", http.StatusBadRequest)
 			return
 		}
 		pj, err := kc.GetProwJob(name)
```
Comment (BenTheElder) [inline prow/cmd/deck/main.go]: FWIW this probably still has a timing attack but:
- the pod names are a bit long
- knowing the pod names shouldn't matter if we are sure to filter out access to them everywhere

**Comment by BenTheElder** [comment]
On a second pass:
I think it might be a better plan long-term to wrap the prowJob client with one that filters in the client so that all job acquiring endpoints are filtered in the same way reliably. 
It might even be worth just sticking this in the base client and injecting a config agent into all of these.

**Comment by cjwagner** [comment]
@BenTheElder Ok, I'll make that change. The log endpoints are safe as is because they look up jobs in the `JobAgent`s `jobsIDMap` which will not include any secret prowjobs.

**Comment by cjwagner** [comment]
@BenTheElder The `kube.Client` that fetches prowjobs does the filtering now. PTAL.

**Comment by BenTheElder** [comment]
Thanks!

cc @tallclair: this should prevent prowjobs in the security repo from being queried in any way from our public dashboard (`deck`) web endpoints at [prow.k8s.io](http://prow.k8s.io/)

**Pull request merged**